can_read_write.py

```diff
@@ -0,0 +1,94 @@
+# This scipt must run on Pi startup
+
+import can
+import time
+import os
+import sys
+import json
+from threading import Thread
+
+# info handled; specified by https://docs.google.com/document/d/1kUU54jQZAB9nwCM-iA96Kj0fXJ6RNw9nAcBffzji5cU/edit
+# full list of PIDs at https://en.wikipedia.org/wiki/OBD-II_PIDs
+PIDS = {'VEHICLE_CHARGE': 0x2F,
+        'VEHICLE_SPEED': 0x0D,
+}
+
+
+PID_REQUEST = 0x7DF
+PID_REPLY = 0x7E8
+
+OUTPUT_PATH = os.path.join(sys.path[0], 'data.json')
+
+
+class FilteredBufferReader(can.BufferedReader):
+    def __init__(self):
+        can.BufferedReader.__init__(self)
+
+    def on_message_received(self, msg):
+        if msg.arbitration_id == PID_REPLY and msg.data[2] in PIDS.keys():
```

Comment:
Obviously this isn't anything we need to worry about now, but in the future we'll have our 4G hotspot on the CAN bus that will also need to request sensor data. Just checking if the `arbitration_id == PID_REPLY` could then result in the Pi considering data that it didn't actually request.

I'm not sure if that's actually an issue, but it could lead to some unintended side-effects (like if the Pi data was supposed to update at a different frequency from the data requested by the 4G hotpot.

Reply:
This could be an issue depending on how we implement the hotspot -- I'll keep an eye on this.